optuna/study/_multi_objective.py

```diff
@@ -127,102 +80,128 @@ def _fast_non_dominated_sort(
     Plus, the algorithm terminates whenever the number of sorted trials reaches n_below.
 
     Args:
-        objective_values:
-            Objective values of each trials.
+        loss_values:
+            Objective values, which is better when it is lower, of each trials.
         penalty:
             Constraints values of each trials. Defaults to None.
         n_below: The minimum number of top trials required to be sorted. The algorithm will
             terminate when the number of sorted trials reaches n_below. Defaults to None.
 
     Returns:
         An ndarray in the shape of (n_trials,), where each element is the non-dominated rank of
-        each trial. The rank is 0-indexed and rank -1 means that the algorithm terminated before
-        the trial was sorted.
+        each trial. The rank is 0-indexed. All the rank worse than the top-n_below has the same
+        value larger than or equal to n_below.
     """
     if penalty is None:
-        ranks, _ = _calculate_nondomination_rank(objective_values, n_below=n_below)
-        return ranks
+        return _calculate_nondomination_rank(loss_values, n_below=n_below)
 
-    if len(penalty) != len(objective_values):
+    if len(penalty) != len(loss_values):
         raise ValueError(
-            "The length of penalty and objective_values must be same, but got "
-            "len(penalty)={} and len(objective_values)={}.".format(
-                len(penalty), len(objective_values)
-            )
+            "The length of penalty and loss_values must be same, but got "
+            f"len(penalty)={len(penalty)} and len(loss_values)={len(loss_values)}."
         )
-    nondomination_rank = np.full(len(objective_values), -1)
+
+    nondomination_rank = np.full(len(loss_values), -1, dtype=int)
     is_penalty_nan = np.isnan(penalty)
-    n_below = n_below or len(objective_values)
+    n_below = n_below or len(loss_values)
 
     # First, we calculate the domination rank for feasible trials.
     is_feasible = np.logical_and(~is_penalty_nan, penalty <= 0)
-    ranks, bottom_rank = _calculate_nondomination_rank(
-        objective_values[is_feasible], n_below=n_below
+    nondomination_rank[is_feasible] = _calculate_nondomination_rank(
+        loss_values[is_feasible], n_below=n_below
     )
-    nondomination_rank[is_feasible] += 1 + ranks
     n_below -= np.count_nonzero(is_feasible)
 
     # Second, we calculate the domination rank for infeasible trials.
+    top_rank_in_infeasible = np.max(nondomination_rank[is_feasible], initial=-1) + 1
```

Comment:
The rank for infeasible trials seems to be worse than the "worst rank" when `n_below` is specified, is it intended?

Reply:
[rephrase] Is it fine to have `top_rank_in_infeasible` to be worse than the rank after the top `n_below`? Or is it fine to have a non-uniform value after the top `n_below` when `len(is_feasible) > n_below`?
More precisely, `top_rank_in_infeasible` becomes `nondomination_rank[is_feasible] + 1` when `len(is_feasible) > n_below` holds.